### ISSUE TITLE
No way to access dom node of TextInput

### ISSUE THREAD

**Issue by kaitlynbrown** [necolas/react-native-web#1238]
**The problem**
No matter what I do, I can't seem to get a reference to the dom node of a TextInput element

**How to reproduce**

Steps to reproduce:
1. Render either of the following components:
```
class TextInputRefTest extends React.Component {
  constructor(props) {
    super(props);
    this.textInput = React.createRef();
  }

  componentDidMount = () => {
    console.log("textInput:", this.textInput);
  }

  render() {
    return <TextInput ref={this.textInput} />
  }
}
```
```
class TextInputRefTest extends React.Component {
  componentDidMount = () => {
    console.log("textInput:", this.textInput);
  }

  render() {
    return (
      <TextInput ref={el => {
        console.log("setting textInput ref");
        this.textInput = el;
        }}
      />
    )
  }
}
```
2. Observe the logs and see that the ref never gets set.

**Expected behavior**
The logged ref should not be null

**Environment (include versions). Did this work in previous versions?**

* React Native for Web (version): 0.9.9
* React (version): 16.8.0-alpha.1
* Browser: Chrome 72.0.3626.81

**Comment by necolas** [comment]
Using a ref on a component never gives you the underlying platform node. You need to use `findNodeHandle`, although in the future this will no longer be supported and you will use `forwardedRef`. But you might not need to get the DOM nice at all and could use setNativeProps

**Comment by mikeaustin** [comment]
I'm trying to use React DnD with RNW, because other solutions don't work, or are not optimized to use HTML5 DnD. I don't care about compatibility, I just like the RN API.

I've found a few ways to get the DOM element, none are pretty:

```
const FancyBin = React.forwardRef(({ domRef }, ref) => {
  useEffect(() => {
    domRef(ReactDOM.findDOMNode(ref.current))
  })

  return (
    <View ref={ref}>
      <Text>Hello</Text>
    </View>
  )
})

const ref = React.createRef()
const foo = <FancyBin ref={ref} domRef={drop} />
```

I believe the other way was I used onLayout() and used event.nativeEvent.target. Using forwardRef on its own did not give the DOM node – I have to use forwardRef in combination with findDOMNode above.
